crates/sui-faucet/src/main.rs

```diff
@@ -102,10 +102,18 @@ async fn main() -> Result<(), anyhow::Error> {
                     Duration::from_secs(1),
                 ))
                 .concurrency_limit(max_concurrency)
-                .layer(Extension(app_state))
+                .layer(Extension(app_state.clone()))
                 .into_inner(),
         );
 
+    spawn_monitored_task!(async move {
+        loop {
+            // Every 60 seconds we clear the wal coins
+            tokio::time::sleep(Duration::from_secs(60)).await;
```

Comment:
nit: Maybe make this configurable, with a default value? 60s seemed quite often to me initially, but it should be okay.  Even with 500 available coins, and fullnode down, this would only result in an additional 8TPS as we retry these transactions.